www/index.js

```diff
@@ -46,14 +46,16 @@ document.body.appendChild(output);
 const canvas = document.createElement('canvas');
 canvas.id = 'canvas';
 document.body.appendChild(canvas);
-function main() {
-  const result = wasm.draw(
+
+const game = wasm.make_game()
+function drawOneFrame() {
+  const result = game.draw(
       curInput.up, curInput.down, curInput.left, curInput.right);
   if (result == null) {
     output.innerText = 'Failed to draw!';
   } else {
-    output.innerText = 'All is well. (hold up for smile)';
+    output.innerText = 'All is well.';
   }
-  requestAnimationFrame(main);
+  requestAnimationFrame(drawOneFrame);
```

Comment:
Is this the only way to re-enter WASM? Is there no way to do this within Rust?

Reply:
We'd need the ability to send a closure from Rust to the JS bindings to register the animation frame. It's doable, just another thing to learn